privacyidea/lib/utils.py

```diff
@@ -849,11 +849,19 @@ def hash_password(password, hashtype):
         return "{SSHA512}" + pw
     elif hashtype == "OTRS":
         password = hashlib.sha256(password).hexdigest()
+    elif hashtype == "MD5CRYPT":
+        salt = geturandom(4, True)
+        password = crypt.crypt(password, "$1$" + salt + "$")
+    elif hashtype == "SHA512CRYPT":
+        salt = geturandom(4, True)
```

Comment:
Why wouldn't we add a bit more salt? (I like salty ;-)

Reply:
You are right, up to 16 bytes are supported by crypt(3), while I had exactly 8 bytes in mind. I'll double that.